src/gui/qgsmapcanvassnappingutils.cpp

```diff
@@ -22,7 +22,7 @@
 #include <QProgressDialog>
 
 QgsMapCanvasSnappingUtils::QgsMapCanvasSnappingUtils( QgsMapCanvas *canvas, QObject *parent )
-  : QgsSnappingUtils( parent, QgsSettings().value( QStringLiteral( "/qgis/digitizing/snap_invisible_feature" ), false ).toBool() )
+  : QgsSnappingUtils( parent, QgsSettings().value( QStringLiteral( "/qgis/digitizing/snap_invisible_feature" ), false ).toBool(), true )
```

Comment:
I think this also breaks existing API - as can be seen in the required changes to unit tests - the default should probably stay as non-asynchronous, but in QGIS app initialization the asynchronous mode could be set.